platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/stats/ReconnectMapMetrics.java

```diff
@@ -67,48 +67,77 @@ public ReconnectMapMetrics(
                         .withDescription("number of internal node hashes transferred"));
-        this.internalCleanHashes =
-                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalCleanHashes", label))
+        this.internalCleanHashes = metrics.getOrCreate(
+                new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("internalCleanHashes", label))
                         .withDescription("number of clean internal node hashes transferred"));
         this.internalData =
-                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalData", label))
+                metrics.getOrCreate(new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("internalData", label))
                         .withDescription("number of internal node data transferred"));
         this.internalCleanData =
-                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("internalCleanData", label))
+                metrics.getOrCreate(new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("internalCleanData", label))
                         .withDescription("number of clean internal node data transferred"));
 
         this.leafHashes =
-                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafHashes", label))
+                metrics.getOrCreate(new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("leafHashes", label))
                         .withDescription("number of leaf node hashes transferred"));
         this.leafCleanHashes =
-                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafCleanHashes", label))
+                metrics.getOrCreate(new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("leafCleanHashes", label))
                         .withDescription("number of clean leaf node hashes transferred"));
-        this.leafData = metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafData", label))
+        this.leafData = metrics.getOrCreate(new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("leafData", label))
                 .withDescription("number of leaf node data transferred"));
         this.leafCleanData =
-                metrics.getOrCreate(new Counter.Config(RECONNECT_MAP_CATEGORY, formatName("leafCleanData", label))
+                metrics.getOrCreate(new LongGauge.Config(RECONNECT_MAP_CATEGORY, formatName("leafCleanData", label))
                         .withDescription("number of clean leaf node data transferred"));
+
+        // Reset metric values to zeros on reconnect start
+        resetMetrics();
     }
 
     private static String formatName(final String name, final String label) {
         return (label == null || label.isBlank() ? name : (name + "_" + label + "_")) + "Total";
     }
 
+    private static void add(final LongGauge metric, final long value) {
+        metric.set(metric.get() + value);
```

Comment:
this is not thread safe, 2 threads updating at the same time might increment it only once. I don't know if there will be any contention on this, but its worth noting.

@hendrikebbers this PR points to the fact that there is a requirement that we cannot meet atm. is there any way to support it in the short term?

Reply:
Yes, that's correct, this method is not thread-safe. Currently, the reconnect is a single-threaded process and its protocol is not designed to handle reconnect of multiple sub-trees in parallel today. While there are some ideas to parallelize the process in the future, there's not any concrete plans for doing that in the short term.

So as far as the current usages of the ReconnectMapMetrics are concerned, it's okay for this method to not be thread-safe at this time.

If in the future a parallel reconnect algorithm is implemented, then this metrics collection code would need changes to synchronize access to the metric values. But doing so is out of scope of the current fix for the reasons mentioned above.